cmd/dsl/main.go

```diff
@@ -7,33 +7,72 @@ import (
 	"errors"
 	"flag"
 	"fmt"
-	"github.com/ssb-ngi-pointer/netsim/internal/parser"
-	"golang.org/x/sync/errgroup"
 	"io"
 	"log"
 	"net"
 	"os"
 	"os/exec"
 	"os/signal"
 	"path/filepath"
+	"runtime"
+	"sort"
 	"strconv"
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/ssb-ngi-pointer/netsim/internal/parser"
+	"golang.org/x/sync/errgroup"
 )
 
+type Process struct {
+	cmd     *exec.Cmd
+	logfile *os.File
+}
+
 type Puppet struct {
-	Port       int
-	directory  string
-	feedID     string
-	name       string
-	caps       string
-	hops       int
-	seqno      int64
-	secretDir  string
-	omitOffset bool
+	directory     string
+	feedID        string
+	name          string
+	caps          string
+	secretDir     string
+	omitOffset    bool
+	port          int
+	hops          int
+	seqno         int
+	totalMessages int
+	totalTime     time.Duration
+	slept         time.Duration
+	lastStart     time.Time
+	process       Process // holds cmd & logfile of a running puppet process
+}
+
+func (p Puppet) String() string {
+	return fmt.Sprintf("[%s@%d] %s", p.name, p.seqno, p.feedID)
+}
 
-	stopProcess context.CancelFunc
+func (p *Puppet) stopTimer() {
+	if p.lastStart.IsZero() {
+		return
+	}
+	p.totalTime += time.Since(p.lastStart)
+	var zero time.Time
+	p.lastStart = zero
+}
+
+func (p *Puppet) countMessages() {
+	seqnos, err := queryLatest(p)
```

Comment:
suggest to handle this error.

A new impl might not support the call just yet and you wouldn't know other then the number staying zero for some reason.

Reply:
v good point!

Reply:
doën